docs/Configuration/Dialplan/Pattern-Matching.md

```diff
@@ -3,331 +3,255 @@ title: Pattern Matching
 
 When Alice dials a number on her phone, Asterisk first looks for an extension (in the context specified by the channel driver configuration) that matches exactly what Alice dialed. If there's no exact match, Asterisk then looks for a pattern that matches. After we show the syntax and some basic examples of pattern matching, we'll explain how Asterisk finds the best match if there are two or more patterns which match the dialed number.
 
-Special Characters Used in Pattern Matching
-===========================================
+**Pattern matches always begin with an underscore**. This is how Asterisk recognizes that the extension is a pattern and not just an extension with a funny name. Within the pattern, we use various letters and characters to represent a *family* of dial numbers.
 
-Pattern matches always begin with an underscore. This is how Asterisk recognizes that the extension is a pattern and not just an extension with a funny name. Within the pattern, we use various letters and characters to represent sets or ranges of numbers. Here are the most common letters:
 
-X
--
+Character Set
+-------------
 
-The letter **X** or **x** represents a single digit from 0 to 9.
+Place the characters you want to match between square brackets. If you want to match an `a` or an `e`, use `[ae]`, like in `gr[ae]y`, which matches either `gray` or `grey`.
 
-Z
--
+**Examples**:
+* `_gr[ae]y`: Matches `gray`, `grey`
+* `_[1-468]`: Matches `1`, `2`, `3`, `4`, `6`, `8`. It does not match any number from one to four hundred sixty-eight!
+* `_a[c-e]`: Matches `ac`, `ad`, `ae`
+* `_[127-9ac-e]`: Matches `1`, `2`, `7`, `8`, `9`, `a`, `c`, `d`, `e`
 
-The letter **Z** or **z** represents any digit from 1 to 9.
 
-N
--
+Special Symbols
+---------------
 
-The letter **N** or **n** matches any digit from 2-9.
+Some Character Sets can be written in a short form. Here are the most common ones:
 
+* `_[0-9]` = `_X` (or `x`): Represents a single digit from 0 to 9.
+* `_[1-9]` = `_Z` (or `z`): Represents any digit from 1 to 9.
+* `_[2-9]` = `_N` (or `n`): Matches any digit from 2 to 9.
 
+**Examples**:
+* `_6[34]XX`: Short form of `_64XX` and `_63XX`, matches `6400`, `6300`, `6401`, `6301`, `6420`, `6320`, `6433`, `6333`, ...
+* `_6[34X]`: Matches `63`, `64`, `6X`. The `X`, `N`, and `Z` convenience notations have no special meaning within a set, they represent the literal character.
 
-Ranges used in Pattern Matching
-===============================
-
-A numeric range can be used to match against a dialed number. This is also called a Character Set
-
+!!! note
+    The only characters with special meaning within a set are the `-` character, to define a range between two characters, the  `\` character to escape a special character available within a set, and  
```

Comment:
What are the other character with special meaning? Notice the final and ...

Reply:
> The only characters with special meaning within a set are the `-` character, to define a range between two characters, the `\` character to escape a special character available within a set, and the `]` character which closes the set. The treatment of the `\` character in pattern matching is somewhat haphazard and may not escape any special character meaning correctly.